compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirConflictingProjectionChecker.kt

```diff
@@ -44,10 +48,8 @@ object FirConflictingProjectionChecker : FirBasicDeclarationChecker() {
     }
 
     private fun checkTypeRef(typeRef: FirTypeRef, context: CheckerContext, reporter: DiagnosticReporter) {
-        // TODO: remaining implicit types should be resolved as an error type, along with proper error kind,
-        //  e.g., type mismatch, can't infer parameter type, syntax error, etc.
-        val declaration = typeRef.safeAs<FirResolvedTypeRef>()
-            ?.coneTypeSafe<ConeClassLikeType>()
+        val declaration = typeRef.coneType
```

Comment:
Use `typeRef.coneTypeUnsafe<ConeClassLikeType>()`

Reply:
I assumed you meant `...coneTypeSafe...` since here we can still see some resolved type other than class-like type. Using `...coneTypeUnsafe...` simply fails a lot.